js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -18,7 +17,38 @@ import {
   OpenInferenceSpanKind,
   SemanticConventions,
 } from "@arizeai/openinference-semantic-conventions";
-import { GenericFunction, SafeFunction } from "./types";
+import {
+  GenericFunction,
+  SafeFunction,
+  ObjectWithModel,
+  RetrieverQueryEngineQueryMethodType,
+  RetrieverRetrieveMethodType,
+  QueryEmbeddingMethodType,
+} from "./types";
+import { BaseEmbedding, BaseRetriever } from "llamaindex";
+
+/**
+ * Checks if
+ * @param prototype -
+ * @returns
+ */
+export function isRetriever(
+  moduleClassPrototype: unknown,
+): moduleClassPrototype is BaseRetriever {
+  return (
+    !!moduleClassPrototype && !!(moduleClassPrototype as BaseRetriever).retrieve
```

Comment:
casting here doesn't make this safer. Let me pull down and see if I can make it a bit more tolerant to bad inputs. We should cover this with tests.